contracts/stork_chainlink_adapter/contracts/StorkChainlinkAdapter.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.8.24 <0.9.0;
+
+import "./IStorkTemporalNumericValueUnsafeGetter.sol";
+import "./StorkStructs.sol";
+
+/**
+ * @title A port of the ChainlinkAggregatorV3 interface that supports Stork price feeds
+ */
+contract StorkChainlinkAdapter {
+    bytes32 public priceId;
+    IStorkTemporalNumericValueUnsafeGetter public stork;
+
+    constructor(address _stork, bytes32 _priceId) {
+        priceId = _priceId;
+        stork = IStorkTemporalNumericValueUnsafeGetter(_stork);
+    }
+
+    function decimals() external pure returns (uint8) {
+        return 18;
+    }
+
+    function description() public pure returns (string memory) {
+        return "A port of a chainlink aggregator powered by Stork";
+    }
+
+    function version() public pure returns (uint256) {
+        return 1;
+    }
+
+    function latestAnswer() public view virtual returns (int256) {
+        return stork.getTemporalNumericValueUnsafeV1(priceId).quantizedValue;
+    }
+
+    function latestTimestamp() public view returns (uint256) {
+        return stork.getTemporalNumericValueUnsafeV1(priceId).timestampNs;
+    }
+
+    function latestRound() public view returns (uint256) {
+        // use timestamp as the round id
+        return latestTimestamp();
+    }
+
+    function getAnswer(uint256) public view returns (int256) {
+        return latestAnswer();
+    }
+
+    function getTimestamp(uint256) external view returns (uint256) {
+        return latestTimestamp();
+    }
+
+    /*
+    * @notice This is exactly the same as `latestRoundData`, just including for parity with Chainlink
+    * Stork doesn't store roundId on chain so there's no way to access old data by round id
+    */
+    function getRoundData(
```

Comment:
other option here is to always fail if people try to call getRoundData since we won't be exposing old data, but that could cause problems with apps that are currently using this function on the chainlink aggregator.